src/trio/_core/_tests/test_thread_cache.py

```diff
@@ -195,3 +196,39 @@ def deliver(_: object) -> NoReturn:
     err = capfd.readouterr().err
     assert "don't do this" in err
     assert "delivering result" in err
+
+
+@pytest.mark.skipif(not hasattr(os, "fork"), reason="os.fork isn't supported")
+def test_clear_thread_cache_after_fork() -> None:
+    assert hasattr(os, "fork")
+
+    def foo() -> None:
+        pass
+
+    # ensure the thread cache exists
+    done = threading.Event()
+    start_thread_soon(foo, lambda _: done.set())
+    done.wait()
+
+    child_pid = os.fork()
+
+    # try using it
+    done = threading.Event()
+    start_thread_soon(foo, lambda _: done.set())
+    done.wait()
+
+    if child_pid != 0:
+        # if this test fails, this will hang, triggering a timeout.
+        os.waitpid(child_pid, 0)
+    else:
+        try:
+            import coverage
+
+            cov = coverage.Coverage.current()
+            if cov:  # pragma: no branch
```

Comment:
can add a comment+reference to the coverage.py issue why the pragmas are "needed"

Reply:
+1